File: abc.com/src/main/java/application/phase/Phase.java
package application.phase;

import javax.validation.constraints.NotNull;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import application.common.BasicInfo;

@Document
public class Phase  extends BasicInfo{
	
	@Id
	private String id;
	@NotNull
    private String name;
	
	public Phase() {
		
	}
	
	public Phase(String name) {
		this.name = name;
	}
	public String getId() {
		return id;
	}
	public void setId(String id) {
		this.id = id;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	
	
}

File: abc.com/client/src/app/phase/phase.ts
import { Class } from '@angular/core';

export class Phase {

  private  id: string;
  private  name: string;
  
  constructor() {
  }
  
   public setId ( id: string ) {
    this.id = id;
  }
  public getId (): string {
    return this.id ;
  }
  
  public setName ( name: string ) {
    this.name = name;
  }
  
  public getName (): string {
    return this.name ;
  }
}
